lib/src/lints/function_lines_of_code/function_lines_of_code_rule.dart

```diff
@@ -50,15 +50,25 @@ class FunctionLinesOfCodeRule
   ) {
     void checkNode(AstNode node) => _checkNode(resolver, reporter, node);
 
-    context.registry.addDeclaration((declarationNode) {
-      final isIgnored = config.parameters.exclude.shouldIgnore(declarationNode);
+    void checkDeclarationNode(Declaration node) {
+      final isIgnored = config.parameters.exclude.shouldIgnore(node);
+      if (isIgnored) {
+        return;
+      }
+      checkNode(node);
+    }
 
-      if (isIgnored) return;
+    // Check for an anonymous function
+    void checkFunctionExpressionNode(FunctionExpression node) {
+      if (node.parent is FunctionDeclaration) {
```

Comment:
Add comment on why this check is here:
```suggestion
      // Prevent checking named functions twice
      if (node.parent is FunctionDeclaration) {
```